### PR TITLE
Pr-04-06

### PR DESCRIPTION
**Pull request by 6gunner**
1. Add test code about signing & verifying;
2. Update @openzeppelin/contract version;
3. Fix some original bugs that result in running errors.

### CODE REVIEW

File: basic/23-erc865-and-erc875/testAccounts.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[
```
Comment (maroonstar) [inline basic/23-erc865-and-erc875/testAccounts.json:1]: Put these into .env file would be better I think.

**Comment by maroonstar** [comment]
That's Great !

File: basic/23-erc865-and-erc875/testAccounts.json (hunk 1 above)
Comment (6gunner) [reply basic/23-erc865-and-erc875/testAccounts.json:1]: yeah, but need to do some string operations cause there are many privates keys.

File: basic/23-erc865-and-erc875/Sign&Verify.md [hunk 2]
```diff
@@ -0,0 +1,608 @@
...
+
+
+
+### 二、signTypedData_v1
+
+v1基于早期的EIP-712的协议，对数据的格式要求如下：
+
+```
+{
+	  type: 'array',
```
Comment (luckyyang) [inline basic/23-erc865-and-erc875/Sign&Verify.md:294]: The format is messed up

File: basic/23-erc865-and-erc875/Sign&Verify.md [hunk 3]
```diff
@@ -0,0 +1,608 @@
...
+    'value': 42,
+  },
+];
+// 找不到了
+const signature = await web3.personal.signTypedData(typedData);
+```
+
+如果想尝试，可以用matamask提供的`eth_signTypedData`方法来唤起。效果如下：
+
+<img src="https://ipic-coda.oss-cn-beijing.aliyuncs.com/20220325/image-20220325103103575.png" alt="image-20220325103103575" style="zoom:50%;" />
```
Comment (luckyyang) [inline basic/23-erc865-and-erc875/Sign&Verify.md:333]: Image can not be loaded

**Review by luckyyang — COMMENTED** [review]
Awesome work! Need to do several small changes

File: basic/23-erc865-and-erc875/Sign&Verify.md (hunk 3 above)
Comment (6gunner) [reply basic/23-erc865-and-erc875/Sign&Verify.md:333]: It is loaded normally for me. But I will replace it with local image

**Review by liberhe — DISMISSED** [review]

**Review by maroonstar — APPROVED** [review]

**Review by liberhe — APPROVED** [review]

**Pull request merged**